packages/@extension-kid/toast-notifications/components/ToastContainer.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from "react";
+import { Container } from "inversify";
+import { combineLatest, of } from "rxjs";
+import { catchError, startWith, map } from "rxjs/operators";
+
+import { Toaster } from "@dcos/ui-kit";
+import { componentFromStream } from "@dcos/data-service";
+
+import { ToastNotification } from "../ToastNotification";
+import ToastExtension from "../ToastExtension";
+import {
+  NotificationService,
+  NotificationServiceType
+} from "@extension-kid/notification-service";
+import makeToast from "./Toast";
+
+interface ToastContainerProps {
+  className: string;
+  primaryActionClassName?: string;
+  secondaryActionClassName?: string;
+}
+
+function makeToasts(
+  toasts: ToastNotification[],
+  extension: ToastExtension,
+  primaryActionClassName: string | undefined,
+  secondaryActionClassName: string | undefined
+) {
+  return toasts.map(tn =>
+    makeToast(tn, extension, primaryActionClassName, secondaryActionClassName)
+  );
+}
+
+function makeToastContainer(
+  container: Container
+): React.ComponentType<ToastContainerProps> {
+  return componentFromStream<ToastContainerProps>(props$ => {
+    const ns = container.get<NotificationService>(NotificationServiceType);
+    const extension = ns.findExtension(ToastNotification.NotificationType);
+    const toastExtension =
+      extension !== undefined ? (extension as ToastExtension) : undefined;
+
+    if (!toastExtension) {
+      return of(<div />);
+    }
+    return combineLatest<[ToastContainerProps, ToastNotification[]]>([
+      props$,
+      toastExtension.Toast$
+    ]).pipe(
+      map(([props, toasts]) => {
+        return (
+          <div className={props.className}>
+            <Toaster>
+              {toasts.length > 0
+                ? makeToasts(
```

Comment:
i'm not sure just from looking at it, but could we also expose `makeToast` directly, so we could do a `toasts.map(t => makeToast(t, /*...*/))` here that would yield `[]` automatically when no toasts are present instead of having a conditional on `length`? (to me it feels like `makeToasts` is a worse abstraction than `makeToast`)

Reply:
yea `makeToasts` is redundant at this point. I'll refactor it out.